feat: Add native lima Docker provider support (#5892)
ddev/ddev: docs/content/users/install/docker-installation.md

    !!!warning "Colima can only work in your home directory unless you do further configuration"
        By default, Colima only mounts your home directory, so it’s easiest to use it in a subdirectory there. See the `~/.colima/default/colima.yaml` for more information, or notes in [colima.yaml](https://github.com/abiosoft/colima/blob/fc948f8f055600986f87e29e3e632daf56ac8774/embedded/defaults/colima.yaml#L130-L143).

    ### Lima

>     [Lima](https://github.com/lima-vm/lima) is a free and open-source project supported by the [Cloud Native Computing Foundation](https://cncf.io/).

Author: Sorry, can't understand your question but feel free to propose something.

>     1. Install Lima with `brew install lima`.

Author: I like it better this way, didn't like the other one as much.

    2. Install the `docker` client if you don't already have it with `brew install docker`.

>     3. Create a default Lima setup with 4 CPUs, 6GB memory, 100GB storage, and Cloudflare DNS, adjusting as needed:

Reviewer: ```suggestion
    3. Create a 100GB VM in Lima with 4 CPUs, 6GB memory, and Cloudflare DNS. Adjust to your own needs:
```

Reviewer: Also oriented to the proposed changes on the colima quickstart PR

    ```

>     limactl create --vm-type=vz --mount-type=virtiofs --mount-writable --memory=6 --cpus=4 --disk=100 template://docker

Reviewer: i am cautious and a bit afraid of mounting the entire home directory, most of the lima maintainers share the same worry. That is the reason why out of the box only the lima temp directory is added. but the rancher maintainer is the only lima maintainer who mounts his entire homedirectory for convenience reasons. just as a note. i am ok with it but personally i only use --mount="~/Sites:w" (with Sites being the directory i store all my projects)

Author: I understand that your caution, but this is the default behavior and doing it otherwise creates enormous support problems.

>     docker context create lima-default --docker "host=unix://$HOME/.lima/default/sock/docker.sock"

Reviewer: it probably should be noted that this command has only to be run once after the initial creation of the VM. and this line as well as the next (docker context use) are also output by default. they are part of the `message`that is defined in the instances yaml file.

Author: This is shown as part of the first-time setup.

    docker context use lima-default
    ```

>     After the initial run above, you can use `limactl start`.  Run `limactl list` to see configured setup.

Reviewer: ```suggestion
    After the initial setup, you can start your instance with `limactl start`. Run `limactl --help` to see the list of available commands.
```
instead of adding the command to see the configured setup i think providing the help command to get the list of all available command might be more helpful (which also includes the list command)

Author: I like it as is. People have to branch out to learn more.

    When your computer restarts, you’ll need to `limactl start` again.

    !!!warning "Docker contexts let the Docker client point at the right Docker server"
        The Docker provider you're using is selected with `docker context`. You can see the available contexts with `docker context ls` and the currently selected one with `docker context show`. With the setup above you'll want `docker context use lima-default`.

>     !!!warning "Lima only mounts filesystems in your home directory unless you do further configuration"

Reviewer: same as for the colima quickstart. i think it would make sense to clarify that those mounts are not necessary for lima to run but for ddev projects to run?

Author: Here we're only interested in people running DDEV projects. People can do other things with Lima if they want to. 

If people don't put their projects on a filesystem that Lima mounts then DDEV is nonfunctional. Most people put their projects somewhere in the homedir.
